cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ConnectionHandler.java

```diff
@@ -155,6 +165,8 @@ public void getConnection(HttpServiceRequest request, HttpServiceResponder respo
                             "Getting connection in system namespace is currently not supported");
         return;
       }
+      contextAccessEnforcer.enforce(new SystemAppEntityId(namespace, "pipeline", "connection",
+                                                          connection), StandardPermission.GET);
```

Comment:
here the connection is the name, when you construct the entity id, please use connection id, you can get it with `new ConnectionId(ns, connection).getConnectionId`, the reason for this is the connection name can contain all characters including space, @ , etc. These characters might be forbidden in many auth enforcers. So let's use id which only allow _ and alphanumeric characters

Reply:
This applies to all places where enforcement happens

Reply:
ack